json_serializable/lib/src/utils.dart

```diff
@@ -46,7 +46,19 @@ void throwUnsupported(FieldElement element, String message) =>
 
 FieldRename _fromDartObject(ConstantReader reader) => reader.isNull
     ? null
-    : FieldRename.values[reader.objectValue.getField('index').toIntValue()];
+    : FieldRename.values.singleWhere(
+        (thing) {
+          var name = thing.toString().split('.')[1];
+          var result = reader.objectValue.getField(name);
+          return result != null;
+        },
+        // TODO: remove once pkg:analyzer < 0.35.0 is no longer supported
```

Comment:
`pkg:analyzer` changed how `DartObject` looks when it represents an enum value. I did this here and above so we don't have to bump the pkg:analyzer dep right away w/ this release.

I did test back to `0.33.3` to make sure this works across the supported version range.

Reply:
I rolled this into one helper in the latest commit